pkg/registry/servicecatalog/broker/strategy.go

```diff
@@ -117,3 +127,29 @@ func (brokerRESTStrategy) ValidateUpdate(ctx kapi.Context, new, old runtime.Obje
 
 	return scv.ValidateBrokerUpdate(newBroker, oldBroker)
 }
+
+func (brokerStatusRESTStrategy) PrepareForUpdate(ctx kapi.Context, new, old runtime.Object) {
+	newBroker, ok := new.(*sc.Broker)
+	if !ok {
+		glog.Warning("received a non-broker object to update to")
+	}
+	oldBroker := old.(*sc.Broker)
+	if !ok {
```

Comment:
this is testing the `ok` defined on the previous type-assertion, not this one. change `oldBroker := ...` to `oldBroker, ok := ...` and you should be good